src/components/internal/Modals/UserSettingsModal.component.tsx

```diff
@@ -372,86 +329,16 @@ export class UserSettingsModalComponent extends Component<Props, State> {
         };
     };
 
-    private onProfileImageChange = async (
-        event: ChangeEvent<HTMLInputElement>,
-    ) => {
-        const files = event.target.files;
-        if (!files || files.length < 1) {
-            return;
-        }
-
-        const file = files[0];
-        const previewImageUrl = URL.createObjectURL(file);
-        const image = await loadImage(previewImageUrl);
-        const ratio = image.width / image.height;
-        if (ratio !== 1) {
-            alert('Profile image need to be a square');
-            return;
-        }
-
-        if (image.width > 800) {
-            alert(
-                'Profile image need to be smaller than or equal to 800 x 800',
-            );
-            return;
-        }
-
-        this.setState({
-            isPreviewingProfileImage: true,
-            previewProfileImageUrl: previewImageUrl,
-            profileUploadProgress: {
-                newBytesUploaded: 0,
-                totalPercentage: 0,
-            },
-        });
-
-        const fileUploadSessionId =
-            await this.stateSyncer.createUserProfileUpdateSession();
-        const uploadSession =
-            this.fileUploadSessionFactory.createFileUploadSession(
-                fileUploadSessionId,
-            );
-        const onNewProgressCh = uploadSession.subscribeNewProgress();
-        (async () => {
-            while (true) {
-                const progress = await onNewProgressCh.pop();
-                if (progress === undefined) {
-                    return;
-                }
-
-                this.setState({
-                    profileUploadProgress: progress,
-                });
-            }
-        })().then();
-        const error = await uploadSession.uploadFile(file);
-        if (error) {
-            alert('Fail to upload profile image');
-        } else {
-            await this.stateSyncer.finishUserProfileUpdateSession(
-                fileUploadSessionId,
-            );
-            await this.stateSyncer.pullCurrentUser();
-            const user = this.graphSource.currentUser();
-            this.setState({
-                user,
-            });
-        }
-
+    private async handleFileUploadFinished(fileUploadSessionId: number) {
```

Comment:
```suggestion
    private async handleFileUploadFinished = (fileUploadSessionId: number) => {
```
Let's use `=>` so that scope for createUserProfileUpdateSession automatically bind to `this`.